src/main/java/net/dv8tion/jda/internal/utils/config/MetaConfig.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015-2019 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.utils.config;
+
+import net.dv8tion.jda.api.utils.cache.CacheFlag;
+
+import java.util.EnumSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class MetaConfig
+{
+    private static final MetaConfig defaultConfig = new MetaConfig(null, EnumSet.allOf(CacheFlag.class), false, true);
+    private final ConcurrentMap<String, String> mdcContextMap;
+    private final EnumSet<CacheFlag> cacheFlags;
+    private final boolean enableMDC;
+    private final boolean useShutdownHook;
+
+    public MetaConfig(
+            ConcurrentMap<String, String> mdcContextMap,
+            EnumSet<CacheFlag> cacheFlags, boolean enableMDC, boolean useShutdownHook)
+    {
+        this.cacheFlags = cacheFlags == null ? EnumSet.allOf(CacheFlag.class) : cacheFlags;
+        this.enableMDC = enableMDC;
+        if (enableMDC)
+            this.mdcContextMap = mdcContextMap == null ? new ConcurrentHashMap<>() : null;
+        else
+            this.mdcContextMap = null;
+        this.useShutdownHook = useShutdownHook;
+    }
+
+    public ConcurrentMap<String, String> getMdcContextMap()
+    {
+        return mdcContextMap;
+    }
+
+    public EnumSet<CacheFlag> getCacheFlags()
+    {
+        return cacheFlags;
+    }
+
+    public boolean isEnableMDC()
+    {
+        return enableMDC;
+    }
+
+    public boolean isUseShutdownHook()
+    {
+        return useShutdownHook;
+    }
+
+    public static MetaConfig getDefault()
+    {
+        return defaultConfig;
```

Comment:
Because this is the only one (for JDA instances) that does not get updated later.

- EventConfig adds/removes listeners
- SessionConfig sets timeouts
- AuthorizationConfig doesn't have a default because that would be nonsense